File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};
        
        Stack<Integer> stack = new Stack<>();
        
        for(int i = 0; i < arr.length; i++){
            if(stack.isEmpty()){
                stack.push(arr[i]);
            }else if(stack.peek() == arr[i]){
                stack.pop();
            }else {
                stack.push(arr[i]);
            }
        }
        
        if(stack.isEmpty()) stack.push(-1);
        
        answer = new int[stack.size()];
        
        for(int j = answer.length - 1; j >= 0; j--) answer[j] = stack.pop();

        
        return answer;
    }
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    answer = []
    
    for i in arr:
        if not answer:
            answer.append(i)
        else:
            if answer[-1] == i:
                answer.pop()
            else:
                answer.append(i)
                
    if not answer:
        answer.append(-1)
    
    return answer